server/app/assets/javascripts/admin_program_statuses_view.ts

```diff
@@ -0,0 +1,46 @@
+import {addEventListenerToElements, assertNotNull} from './util'
+
+class AdminProgramStatusesView {
+  private static STATUS_CHANGE_FORM_SELECTOR = '.cf-status-change-form'
+
+  constructor() {
+    this.registerStatusUpdateFormSubmitListener()
+  }
+
+  private registerStatusUpdateFormSubmitListener() {
+    addEventListenerToElements(
+      AdminProgramStatusesView.STATUS_CHANGE_FORM_SELECTOR,
+      'submit',
+      (event: Event) => this.confirmStatusDefaultChange(event),
+    )
+  }
+
+  private confirmStatusDefaultChange(event: Event) {
+    const statusForm = event.target as HTMLFormElement
+
+    const statusText = assertNotNull(statusForm.dataset.statustext)
+    const dontShow = assertNotNull(statusForm.dataset.dontshow) === 'true'
+    const messagePart = assertNotNull(statusForm.dataset.messagepart)
+    const checked = assertNotNull(
+      document.querySelector(
+        `.default-checkbox-${statusText}`,
```

Comment:
Need to rethink the strategy I used here, since statuses with spaces are not going to work.